config/v1/types_oauth.go

```diff
@@ -204,111 +184,92 @@ type IdentityProviderConfig struct {
 
-	// ldap enables user authentication using LDAP credentials
+	// htpasswd enables user authentication using an HTPasswd file to validate credentials
 	// +optional
-	LDAP *LDAPPasswordIdentityProvider `json:"ldap,omitempty"`
+	HTPasswd *HTPasswdIdentityProvider `json:"htpasswd,omitempty"`
 
 	// keystone enables user authentication using keystone password credentials
 	// +optional
-	Keystone *KeystonePasswordIdentityProvider `json:"keystone,omitempty"`
+	Keystone *KeystoneIdentityProvider `json:"keystone,omitempty"`
 
-	// requestHeader enables user authentication using request header credentials
-	RequestHeader *RequestHeaderIdentityProvider `json:"requestHeader,omitempty"`
-
-	// github enables  user authentication using GitHub credentials
-	// +optional
-	GitHub *GitHubIdentityProvider `json:"github,omitempty"`
-
-	// gitlab enables user authentication using GitLab credentials
-	// +optional
-	GitLab *GitLabIdentityProvider `json:"gitlab,omitempty"`
-
-	// google enables user authentication using Google credentials
+	// ldap enables user authentication using LDAP credentials
 	// +optional
-	Google *GoogleIdentityProvider `json:"google,omitempty"`
+	LDAP *LDAPIdentityProvider `json:"ldap,omitempty"`
 
 	// openID enables user authentication using OpenID credentials
 	// +optional
 	OpenID *OpenIDIdentityProvider `json:"openID,omitempty"`
+
+	// requestHeader enables user authentication using request header credentials
+	RequestHeader *RequestHeaderIdentityProvider `json:"requestHeader,omitempty"`
 }
 
 // BasicAuthPasswordIdentityProvider provides identities for users authenticating using HTTP basic auth credentials
-type BasicAuthPasswordIdentityProvider struct {
+type BasicAuthIdentityProvider struct {
 	// OAuthRemoteConnectionInfo contains information about how to connect to the external basic auth server
 	OAuthRemoteConnectionInfo `json:",inline"`
 }
 
-// RemoteConnectionInfo holds information necessary for establishing a remote connection
+// OAuthRemoteConnectionInfo holds information necessary for establishing a remote connection
 type OAuthRemoteConnectionInfo struct {
 	// url is the remote URL to connect to
 	URL string `json:"url"`
-	// ca is a reference to a ConfigMap containing the CA for verifying TLS connections
-	CA ConfigMapReference `json:"ca"`
-
-	// tlsClientCert references a secret containing the TLS client certificate to present when
-	// connecting to the server.
-	// Looks under the key "tls.cert" for the data unless a lookup key is specified in the secret ref
-	TLSClientCert LocalSecretReference `json:"tlsClientCert"`
-
-	// tlsClientKey references a secret containing the TLS private key for the client certificate
-	// Looks under the key "tls.key" for the data unless a lookup key is specified in the secret ref
-	TLSClientKey LocalSecretReference `json:"tlsClientKey"`
+	// ca is a reference to a config map by name containing the CA for verifying TLS connections.
```

Comment:
I would like us to be more verbose here, so there is no ambiguity about the specific CA usage.
e.g., "ca is a reference to a config map by name containing the CA used as a trust anchor to validate the TLS certificate presented by the remote server."

Reply:
also say PEM-encoded CA